spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/BaseDeleteReachableFilesSparkAction.java

```diff
@@ -103,53 +87,46 @@ public DeleteReachableFiles io(FileIO fileIO) {
 
   @Override
   public DeleteReachableFiles deleteWith(Consumer<String> deleteFunc) {
-    this.removeFunc = deleteFunc;
+    this.deleteFunc = deleteFunc;
     return this;
-
   }
 
   @Override
   public DeleteReachableFiles executeDeleteWith(ExecutorService executorService) {
-    this.removeExecutorService = executorService;
+    this.deleteExecutorService = executorService;
     return this;
   }
 
   @Override
   public Result execute() {
-    Preconditions.checkArgument(io != null, "File IO cannot be null");
```

Comment:
I don't think we need this check any more as the `FileIO` is defaulted all the time. This may be null only if the user explicitly sets it. Since we don't validate such cases in other places, it seemed fine to skip this.

Reply:
I see no reason not to leave it in though.